google/cloud/storage/internal/hash_values.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/hash_values.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+
+std::string Format(HashValues const& values) {
+  if (values.md5.empty()) return values.crc32c;
```

Comment:
Do you think it'd be useful to always include the "crc32c=" and "md5=" prefixes in the output, even if one is empty? It seems like it may be useful to know which one is being printed. Consider always formatting using the `StrCat` call on line 27.

Reply:
These strings are returned by user-visible functions (see `ObjectReadStream::received_hash()/computed_hash()` and their counterparts in `ObjectWriteStream`).  I did not want to change their format without more consideration about any backwards compatibility implications.  I mean, the change you suggest may be harmless or not break anybody, I just wanted to avoid the extra thinking.